pkg/scrape/vrbangers.go

```diff
@@ -28,27 +30,47 @@ func VRBangersSite(wg *sync.WaitGroup, updateSite bool, knownScenes []string, ou
 		sc.Site = siteID
 		sc.HomepageURL = strings.Split(e.Request.URL.String(), "?")[0]
 
-		if e.ChildAttr(`link[rel=shortlink]`, "href") == "" {
-			log.Printf("Skipping %s because it is not a valid scene.", e.Request.URL.String())
+		content_id := strings.Split(strings.Replace(sc.HomepageURL, "//", "/", -1), "/")[3]
+
+		r, _ := resty.R().Get("https://content.vrbangers.com/api/content/v1/videos/" + content_id)
+
+		JsonMetadata := r.String()
+
+		// Sneak Peek 2020 URL will crash the scraper
+		//if not valid scene...
+		if gjson.Get(JsonMetadata, "status.message").String() != "Ok" {
 			return
 		}
-		// Scene ID - get from URL
-		sc.SiteID = strings.Split(e.ChildAttr(`link[rel=shortlink]`, "href"), "?p=")[1]
+
+		//Scene ID - back 8 of the"id" via api response
+		sc.SiteID = strings.TrimSpace(gjson.Get(JsonMetadata, "data.item.id").String()[16:])
 		sc.SceneID = slugify.Slugify(sc.Site) + "-" + sc.SiteID
```

Comment:
Since scene IDs are changing, force updating scenes will duplicate all existing scenes. Also some urls have changed<sup>1</sup>, so those will be duplicated even without force updating. Should there be a migration?

<sup>1</sup> Changed slugs:
```
ayumis-first-time-2 -> ayumis-first-time
caught_watching -> caught-watching
downward_slut -> downward-slut
```

Reply:
There was some discussion over on discord on "to migrate not to migrate". I opted to pull the migration code I had originally put in. What's the "sane" choice here?

Reply:
Maybe this?
1. For all scenes, slugify the last component of the URL, with a special case (:vomiting_face:) for ayumis-first-time and save the result. Or since there are just 3... just fix those 3 explicitly.
2. Fetch the list of all scenes from the API (e.g. `/api/content/v1/videos?page=1&type=videos&sort=latest&show_custom_video=1&bonus-video=1&limit=500`).
3. Using values from the JSON, for each scene `update scenes set scene_id = <back 8 of id> where scene_url like '%/<slug>'`.

Also I should probably check out the DIscord...

Reply:
I'm going to put the migration back in. flush the scenes. My head hurts. If someone else wants to code this, you are more than welcome.  Do we want to be doing a lot of coding in "migrations". Or more than scraping in the scrapers?

Reply:
I gave it a shot [here](https://github.com/alexnsfw/xbvr/commit/dfb6f18dabe4c11aa98c89bc54c06687547f7876), it worked great. Never scraped VRBGay or VRBTrans though, so there might be some missing entries in `slugMapping` for those...

Reply:
nice. I'll run through it. I have VRBTrans scraped, so it will be a good test.

Reply:
good to go